sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -317,11 +315,39 @@ public void captureSession(final @NotNull Session session, final @Nullable Objec
     return envelope.getHeader().getEventId();
   }
 
+  @Override
+  public SentryId captureTransaction(SentryTransaction transaction, Scope scope, Object hint) {
+    Objects.requireNonNull(transaction, "Transaction is required.");
+
+    options
+        .getLogger()
+        .log(SentryLevel.DEBUG, "Capturing transaction: %s", transaction.getEventId());
+
+    Session session = null;
+
+    SentryId sentryId = transaction.getEventId();
+
+    try {
+      final SentryEnvelope envelope = buildEnvelope(transaction, session);
+
+      if (envelope != null) {
+        connection.send(envelope, hint);
+      }
+    } catch (IOException e) {
+      options.getLogger().log(SentryLevel.WARNING, e, "Capturing transaction %s failed.", sentryId);
+
+      // if there was an error capturing the event, we return an emptyId
+      sentryId = SentryId.EMPTY_ID;
```

Comment:
wondering if this method would need to assign the `lastEventId` as well, is a transaction an `event`?

Reply:
Technically it is an event but I am not sure. What's the use case for reading `lastEventId`?

Reply:
@maciejwalkowiak user feedback feature, but I guess transactions won't make sense for this use case.